### PR TITLE
feat: Added Edit networks screen modal

### PR DESCRIPTION
**Pull request by NidhiKJha**
This PR is to add edit Networks Modal

NOTE: Functionality of the modal is not included in this PR

## **Related issues**

Fixes: [2685](https://github.com/MetaMask/MetaMask-planning/issues/2685)

## **Manual testing steps**

1. Run Storybook
2. Check the Edit Networks Modal component, it should look like the figma design
3. This PR doesn't have the functionality implemented

### **Before**

NA

### **After**

Screenshot 2024-07-29 at 1 50 43 PM

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import PropTypes from 'prop-types';
```
Comment (EtherWizard33) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.js:1]: looks like lint is the only check left to make it green
Screenshot 2024-07-29 at 6 40 10 PM

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.stories.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
```
Comment (EtherWizard33) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.stories.js:1]: I was able to run the story, works as expected.


One small thing I noticed is the space around the icon is about the double the one from the screenshot in the corresponding issue.

Other than that, looks great, will circle back tomorrow to review the code.

**Review by EtherWizard33 — CHANGES_REQUESTED** [review]

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      useRequestQueue: true,
+    },
+    activeTab: {
+      origin,
+    },
+  };
+
+  const store = configureStore(state);
+  return renderWithProvider(<EditNetworksModal onClose={jest.fn()} />, store);
+};
```
Comment (EtherWizard33) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.test.js:40]: nit: would be nice to have a line seperating the describe section on line 41, from the util function above

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  const t = useI18nContext();
+  const nonTestNetworks = useSelector(getNonTestNetworks);
+  const testNetworks = useSelector(getTestNetworks);
+  return (
+    <Modal
+      isOpen
+      onClose={() => {
+        onClose();
+      }}
+      className="import-nfts-modal"
```
Comment (EtherWizard33) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.js:28]: this className refers to import-nfts, it probably should be edit-networks?

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            onClose();
+          }}
+        >
+          {t('editNetworksTitle')}
+        </ModalHeader>
+        <Box padding={4}>
+          <Checkbox
+            label={t('selectAll')}
+            isChecked={true}
+            // onClick={() => (allAreSelected() ? deselectAll() : selectAll())}
```
Comment (EtherWizard33) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.js:43]: would be nice to have this click handler as part of this PR, unit testing select all with mock props/local state would then be possible

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        <Box padding={4}>
+          <Text variant={TextVariant.bodyMdMedium}>{t('testnets')}</Text>
+        </Box>
+        {testNetworks.map((network) => (
+          <NetworkListItem
+            name={network.nickname}
+            iconSrc={network?.rpcPrefs?.imageUrl}
+            key={network.id}
+            onClick={() => {
+              console.log(network.id);
```
Comment (EtherWizard33) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.js:67]: would be nice to have this click handler as part of this PR, unit testing select and deselect with mock props/local state would then be possible

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js (hunk 5 above)
Comment (NidhiKJha) [reply ui/components/multichain/edit-networks-modal/edit-networks-modal.js:43]: This PR is just to get the skeleton ready. I will add more tests when we will make it functional

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js (hunk 4 above)
Comment (NidhiKJha) [reply ui/components/multichain/edit-networks-modal/edit-networks-modal.js]: Updated here https://github.com/MetaMask/metamask-extension/pull/26097/commits/59624c3ec6a12be6e6f22970bbe3ace3aa54574a

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.test.js (hunk 3 above)
Comment (NidhiKJha) [reply ui/components/multichain/edit-networks-modal/edit-networks-modal.test.js]: I think it's fine as per the usecase here

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.stories.js (hunk 2 above)
Comment (NidhiKJha) [reply ui/components/multichain/edit-networks-modal/edit-networks-modal.stories.js:1]: nice catch! This component is based on using existing NetworkListItem. Lmk ping @mmragkandala for this

**Review by EtherWizard33 — DISMISSED** [review]
LGTM

File: ui/components/multichain/network-list-item/network-list-item.js [hunk 7]
```diff
@@ -115,7 +115,7 @@ export const NetworkListItem = ({
       onClick={onClick}
     >
       {startAccessory ? (
-        <Box marginInlineEnd={2} marginTop={1}>
+        <Box marginTop={1}>
```
Comment (EtherWizard33) [inline ui/components/multichain/network-list-item/network-list-item.js:118]: a lint error, looks like whitespaces

/home/circleci/project/ui/components/multichain/network-list-item/network-list-item.js
  117:25  error  Replace `(⏎········<Box·marginTop={1}>⏎··········{startAccessory}⏎········</Box>⏎······)` with `<Box·marginTop={1}>{startAccessory}</Box>`  prettier/prettier

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
+import PropTypes from 'prop-types';
+import React from 'react';
```
Comment (salimtb) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.js:2]: can we add unit tests for this new component

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  Checkbox,
+  Text,
+  Box,
+} from '../../component-library';
+import { NetworkListItem } from '..';
+
+export const EditNetworksModal = ({ onClose }) => {
```
Comment (salimtb) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.js:18]: i suggest to add props type here

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+            // isIndeterminate={isIndeterminate}
+          />
+        </Box>
+        {nonTestNetworks.map((network) => (
+          <NetworkListItem
+            name={network.nickname}
+            iconSrc={network?.rpcPrefs?.imageUrl}
+            key={network.id}
+            onClick={() => {
+              console.log(network.id);
```
Comment (salimtb) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.js:54]: should we remove this console log ?

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js (hunk 9 above)
Comment (NidhiKJha) [reply ui/components/multichain/edit-networks-modal/edit-networks-modal.js:18]: It uses NetworkListItem components internally which is not TS compatible. So, it's in JS for now

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js (hunk 8 above)
Comment (NidhiKJha) [reply ui/components/multichain/edit-networks-modal/edit-networks-modal.js:2]: For now, this is a dumb component. I will add unit tests to it when I will make it functional

**Review by salimtb — APPROVED** [review]

**Review by EtherWizard33 — APPROVED** [review]
LGTM
nit:  would be nice to include interaction (select, unselect, selectAll, unselectAll), since it would be state local to the component

**Pull request merged**